### PR TITLE
GitLab 8.12.0

### PR DESCRIPTION
**Pull request by solidnerd**
Hey,

i prepared everything for 8.12.0. I will test it later today. After my tests are finished I will start with the release.

@sameersbn  Can you have a look at this PR and check it. I updated someconfiguration that should be updated from older releases. I hope it doesn't break anything.

### CODE REVIEW

**Comment by lenovouser** [comment]
Awesome!

**Comment by reynold-lariza** [comment]
I'm getting this after applying your changes on your changes :|

Running with gitlab-ci-multi-runner 1.6.0 (01b3ea1)
WARNING: environment is not supported by selected executor and shell
Using Shell executor...
stdin: is not a tty
Running on vm2...
Cloning repository...
Cloning into '/root/builds/188d77d4/0/gitlab/sample-repo'...
fatal: unable to access 'https://gitlab-ci-token:xxxxxxxxxxxxxxxxxxxx@git.<host>/sample-repo.git/': The requested URL returned error: 500
ERROR: Build failed: exit status 1

**Comment by sameersbn** [comment]
On it!

File: assets/runtime/config/gitlabhq/rack_attack.rb [hunk 1]
```diff
@@ -10,13 +10,14 @@
...
+  "#{Rails.application.config.relative_url_root}/unsubscribes/",
+  "#{Rails.application.config.relative_url_root}/import/github/personal_access_token"
 ]
 
 # Create one big regular expression that matches strings starting with any of
 # the paths_to_be_protected.
 paths_regex = Regexp.union(paths_to_be_protected.map { |path| /\A#{Regexp.escape(path)}/ })
+rack_attack_enabled = Gitlab.config.rack_attack.git_basic_auth['enabled']
 
 unless Rails.env.test?
```
Comment (sameersbn) [inline assets/runtime/config/gitlabhq/rack_attack.rb]: Upstream shows that this line should be 

```ruby
unless Rails.env.test? || !rack_attack_enabled
```

Did you intentionally avoid this change?

Comment (solidnerd) [inline assets/runtime/config/gitlabhq/rack_attack.rb]: Yeah I didn't see that I will copy it. Thanks for the hint 👍

Comment (sameersbn) [inline assets/runtime/config/gitlabhq/rack_attack.rb]: Other than that, LGTM!

Comment (sameersbn) [inline assets/runtime/config/gitlabhq/rack_attack.rb]: FYI, please hold on to this release until we update the base image with SSL updates.

Comment (reynold-lariza) [inline assets/runtime/config/gitlabhq/rack_attack.rb]: Guys, there's already a tagged v8.12.1, you might want to check it out: https://gitlab.com/gitlab-org/gitlab-ce/commit/afd43301c016553619b60d8db47537d935aad328

**Comment by reynold-lariza** [comment]
Guys, there's already a tagged v8.12.1, you might want to check it out.

**Comment by solidnerd** [comment]
@reynold-lariza  Thanks for the info. I will prepare `8.12.1` and test this after this MR is merged. So first we need to wait for the base image update, because openssl has new vulnerability that could lead to DoS.

**Comment by Ilya-Kuchaev** [comment]
Any news here?

**Comment by solidnerd** [comment]
Still waiting for base image update. Ping @sameersbn

**Comment by eisnerd** [comment]
I've built on top of xenial to remove further, lower severity vulnerabilities, but I've found that a fresh build of the trusty image with either `trusty-20160819` (last one used) or `trusty-20160923.1` (latest) would be adequate, despite the [report](https://quay.io/repository/eisnerd/ubuntu/image/db873fecb371ca7cd8b1525eeb3adcac2e63f55cde3c862fea44579308af7532?tab=vulnerabilities).

Would it be better to leave openssl out of the base image?

**Comment by evenh** [comment]
There is now two PR's to the Ubuntu base image

- https://github.com/sameersbn/docker-ubuntu/pull/11 (bumps to `ubuntu:xenial-20160923.1`) by @eisnerd 
- https://github.com/sameersbn/docker-ubuntu/pull/12 (bumps to `ubuntu:trusty-20160923.1`)

Hopefully @sameersbn will merge one of them soon enough

**Comment by sameersbn** [comment]
oh. sorry.. on it.

**Comment by sameersbn** [comment]
base image has been updated. thanks for your patience.

**Comment by solidnerd** [comment]
I can handle the next releases. If you want.

**Comment by sameersbn** [comment]
next releases of the base image?

**Comment by solidnerd** [comment]
Nope 😅  I mean the next releases  of GitLab 😄

**Comment by sameersbn** [comment]
yes.. please handle the release :smile:

**Comment by dashersw** [comment]
@solidnerd looking forward to it! :)

**Comment by solidnerd** [comment]
I'm on it I test it and release it after it. I thing this take 30 mins. After this the release will happen. Tomorrow I will test and release every '1.12.X' .

**Comment by dashersw** [comment]
Thanks a lot!

**Comment by solidnerd** [comment]
Happy upgrading everyone 😄

**Pull request merged**